flask/src/main/java/net/corda/gradle/flask/FlaskJarTask.java

```diff
@@ -102,7 +104,8 @@ public FlaskJarTask(@Nonnull ObjectFactory objects) {
         from(getProject().tarTree(LauncherResource.instance), copySpec -> exclude(JarFile.MANIFEST_NAME));
 
         Provider<File> heartbeatJarProvider = getProject().provider(() -> {
-            File dest = new File(getTemporaryDir(), HeartbeatAgentResource.instance.getDisplayName());
+            File tempSubDir = Files.createTempDirectory(getTemporaryDir().toPath(), "heartbeat").toFile();
```

Comment:
this will create a temporary archive for every task execution.. It also came to my mind that I had thought about task concurrency when writing this and actually using an hardcoded filename was not an issue as the temporary directory is
`build/tmp/taskName` and you can't have two `FlaskJarTask` instances with the same name in the same project

Reply:
I also assume you won't run the same `flaskJar` task from 2 different Gradle processes concurrently in the same workspace, since that would also cause trouble when creating the final artifact

Reply:
If we don't need it, then we don't need it.